process/rating/ratingsData.go

```diff
@@ -96,15 +106,74 @@ func NewRatingsData(args RatingsDataArg) (*RatingsData, error) {
 		return nil, err
 	}
 
-	return &RatingsData{
-		startRating:           ratingsConfig.General.StartRating,
-		maxRating:             ratingsConfig.General.MaxRating,
-		minRating:             ratingsConfig.General.MinRating,
-		signedBlocksThreshold: ratingsConfig.General.SignedBlocksThreshold,
-		metaRatingsStepData:   metaRatingStep,
-		shardRatingsStepData:  shardRatingStep,
-		selectionChances:      chances,
-	}, nil
+	ratingData := &RatingsData{
+		startRating:                 ratingsConfig.General.StartRating,
+		maxRating:                   ratingsConfig.General.MaxRating,
+		minRating:                   ratingsConfig.General.MinRating,
+		signedBlocksThreshold:       ratingsConfig.General.SignedBlocksThreshold,
+		metaRatingsStepData:         metaRatingStep,
+		shardRatingsStepData:        shardRatingStep,
+		selectionChances:            chances,
+		nodesSetup:                  args.NodesSetupHandler,
+		ratingsSetup:                ratingsConfig,
+		roundDurationInMilliseconds: args.RoundDurationMilliseconds,
+	}
+
+	args.EpochNotifier.RegisterNotifyHandler(ratingData)
+
+	return ratingData, nil
+}
+
+// EpochConfirmed will be called whenever a new epoch is called
+func (rd *RatingsData) EpochConfirmed(epoch uint32, _ uint64) {
+	// TODO: maybe check if new data is the same and avoid re-computing (old consensus group size = new consensus group size, and so on)
```

Comment:
About this TODO: do we need to call `computeRatingStep` based on the number of records in the config.toml, new `ChainParametersByEpoch` section?
If yes, on the constructor we could have a map with the epoch value and only call the compute once on the constructor, and each time you have the epoch as a key in the map.

Reply:
right. computed all the ratingsSteps on constructor and use the correct one on EpochConfirmed